IDE/QNX/example-server/server-tls.c

```diff
@@ -0,0 +1,392 @@
+/* server-tls.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+/* socket includes */
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <unistd.h>
+
+/* wolfSSL */
+#include <wolfssl/options.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/wolfcrypt/ecc.h>
+#include <wolfssl/wolfcrypt/port/caam/wolfcaam.h> /* functions for blob/cover*/
+
+#define DEFAULT_PORT 11111
+
+#undef  USE_CERT_BUFFERS_256
+#define USE_CERT_BUFFERS_256
+#include <wolfssl/certs_test.h>
+
+/* creates a blob then opens it, returns 0 on success */
+static int test_blob(byte* key, int keySz)
+{
+    int ret, i;
+    byte out[keySz + WC_CAAM_BLOB_SZ];
+    int  outSz;
+    int  keyOutSz;
+
+    byte keymod[16];
+    int  keymodSz = 16;
+
+    /* using a key mod of all 1's */
+    XMEMSET(keymod, 1, keymodSz);
+
+    /* example of creating black blog with key for later */
+    outSz = keySz + WC_CAAM_BLOB_SZ;
+    ret = wc_caamCreateBlob_ex(key, keySz, out, (word32*)&outSz,
+            WC_CAAM_BLOB_BLACK, keymod, keymodSz);
+    if (ret != 0) {
+        printf("Error creating black blob\n");
+        return -1;
+    }
+
+    printf("created black blob :");
+    for (i = 0; i < outSz; i++)
+        printf("%02X,", out[i]);
+    printf("\n");
+
+
+    /* open black blob and overwrite key, this is to show that the opened blob
+     * can be used for the upcoming TLS connection */
+    memset(key, 0, keySz);
+    keyOutSz = keySz;
+    ret = wc_caamOpenBlob_ex(out, outSz, key, (word32*)&keyOutSz,
+            WC_CAAM_BLOB_BLACK, keymod, keymodSz);
+    if (ret != 0) {
+        printf("Error opening black blob\n");
+    }
+
+    return ret;
+}
+
+
+/* example of covering an existing key and then making a black blob with it
+ * note that generating an ECC completely in secure memory already covered can
+ * be done by call wc_ecc_make_key */
+int cover(ecc_key* keyOut, const byte* der, word32 derSz)
+{
+    int ret;
+    ecc_key notSecure;
+    word32  idx = 0;
+
+    /* format bit plus public key x and y parameter */
+    byte x963[65];
```

Comment:
Is this ECC (key size * 2 + 1)?

Reply:
yeah, it is the compressed bit | x | y for use with wc_ecc_export_x963